apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -722,7 +722,7 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
       |> put_query_prefix(keyspace_name)
 
     case Repo.fetch_one(query, consistency: Consistency.device_info(:read)) do
-      device_id when is_binary(device_id) -> {:ok, true}
+      {:ok, device_id} when is_binary(device_id) -> {:ok, true}
       nil -> {:ok, false}
```

Comment:
when you implement the suggestion from the previous comment, I would expect this piece of code to be either

```suggestion
      {:ok, _} -> {:ok, true}
      {:error, :not_found} -> {:ok, false}
```

or

```suggestion
      {:ok, device_id} -> {:ok, device_id != nil}
```

based on whether you decide to go with `safe_fetch_one` or `safe_one`